File: read_excel2.py
import pandas as pd
import os
import sys

files = [
    r'C:\Users\zhaoyang26\Desktop\运营岗-实习转正答辩PPT.xlsx',
    r'C:\Users\zhaoyang26\Desktop\关键岗位.xlsx',
    r'C:\Users\zhaoyang26\Desktop\信息流投放-头条&抖音-2版&3版.xlsx',
    r'C:\Users\zhaoyang26\Desktop\city_level_5_251122.xlsx',
    r'C:\Users\zhaoyang26\Desktop\新建 DOCX 文档.docx',
]

output = []
for f in files:
    fname = os.path.basename(f)
    output.append('=' * 60)
    output.append('FILE: ' + fname)
    output.append('=' * 60)
    try:
        xls = pd.ExcelFile(f)
        output.append('Sheets: ' + str(xls.sheet_names))
        for sheet in xls.sheet_names[:3]:
            df = pd.read_excel(f, sheet_name=sheet, header=None, nrows=22)
            output.append('')
            output.append('--- Sheet: ' + sheet + ' ---')
            # write each row
            for i, row in df.iterrows():
                row_data = []
                for val in row:
                    if pd.isna(val):
                        row_data.append('')
                    else:
                        row_data.append(str(val))
                output.append(' | '.join(row_data[:20]))
    except Exception as e:
        output.append('ERROR: ' + str(e))
    output.append('')

with open(r'C:\Users\zhaoyang26\.qclaw\workspace\excel_summary.txt', 'w', encoding='utf-8') as out:
    out.write('\n'.join(output))
print('Done')

File: read_excel2.js
const XLSX = require('xlsx');

const files = [
    'C:\\Users\\zhaoyang26\\Desktop\\我的公文包\\2024年ovmom\\图书产品部-24年目标拆解-私域V3版.xlsx',
    'C:\\Users\\zhaoyang26\\Desktop\\我的公文包\\X业务线文档\\星火-素养 历史数据.xlsx',
    'C:\\Users\\zhaoyang26\\Desktop\\EM业务线\\EM小学预算.xlsx'
];

files.forEach(f => {
    try {
        const workbook = XLSX.readFile(f);
        console.log('=== ' + f.split('\\').pop() + ' ===');
        console.log('Sheets:', workbook.SheetNames);
        
        workbook.SheetNames.forEach(sheetName => {
            const sheet = workbook.Sheets[sheetName];
            const data = XLSX.utils.sheet_to_json(sheet, {header: 1, defval: ''});
            console.log('\n--- Sheet: ' + sheetName + ' ---');
            data.slice(0, 8).forEach((row, i) => {
                if (row.some(c => c !== '')) {
                    console.log('R' + i + ':', row.slice(0, 8).map(c => String(c).substring(0, 20)));
                }
            });
        });
        console.log('');
    } catch (e) {
        console.log('Error:', e.message);
    }
});
